Assets/_BForBoss/_Core/Scripts/Environment/EnemySpawner.cs

```diff
@@ -75,7 +72,7 @@ private void SetupPool()
         private void Release(EnemyBehaviour behaviour)
         {
             _pool.Release(behaviour);
-            _onEnemyKilled?.Invoke();
+            _waveModel?.OnEnemyKilled?.Invoke();
```

Comment:
Ideally, nobody invokes anything. Since you're not updating `_killCount` here. 
You should call some `IncrementKillCount()`, them model should update the _killCount number. 

and whoever is subscribed to it on the other side get's updated.
I'll give an example below.